app/(golems)/golems/components/MermaidDiagram.tsx

```
@@ -0,0 +1,68 @@
+    return (
+      <div className="bg-[#0d0d0d] border border-[#e5950026] rounded-lg p-4 mb-4 flex items-center justify-center min-h-[100px]">
+        <span className="text-[#8b7355] text-sm">Loading diagram...</span>
+      </div>
+    );
+  }
+
+  return (
+    <div
+      ref={containerRef}
+      className="bg-[#0d0d0d] border border-[#e5950026] rounded-lg p-4 mb-4 overflow-x-auto [&_svg]:max-w-full"
+      dangerouslySetInnerHTML={{ __html: svg }}
```

Biome (2.3.14)

[error] Avoid passing content using the dangerouslySetInnerHTML prop.

Setting content using code can expose users to cross-site scripting (XSS) attacks

(lint/security/noDangerouslySetInnerHtml)